hirundo/_iter_sse_retrying.py

```diff
@@ -44,13 +48,26 @@ def _iter_sse():
             connect_headers["Last-Event-ID"] = last_event_id
 
         with connect_sse(client, method, url, headers=connect_headers) as event_source:
-            for sse in event_source.iter_sse():
-                last_event_id = sse.id
-
-                if sse.retry is not None:
-                    reconnection_delay = sse.retry / 1000
-
-                yield sse
+            try:
+                for sse in event_source.iter_sse():
+                    last_event_id = sse.id
+
+                    if sse.retry is not None:
+                        reconnection_delay = sse.retry / 1000
+
+                    yield sse
+            except SSEError:
+                response = requests.get(
+                    url,
+                    headers=connect_headers,
+                    timeout=READ_TIMEOUT,
+                )
+                yield ServerSentEvent(
+                    event="",
+                    data=response.text,
+                    id=uuid.uuid4().hex,
```

Comment:
what is this id number? is it important? maybe extract it as a meaningful constant

Reply:
It isn't actually important, but it is simply a random UUID. I use it to simulate a fake SSE event here such that I can throw a non-SSE error as-if it were an SSE error. This is because FastAPI's automatic exceptions are thrown as non-SSE responses and I am not entirely sure if I can easily override that (it is probably possible, but I prefer a quick-and-dirty solution for now)